source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -114,6 +114,7 @@ void EnvoyQuicServerStream::encodeTrailers(const Http::ResponseTrailerMap& trail
 void EnvoyQuicServerStream::encodeMetadata(const Http::MetadataMapVector& /*metadata_map_vector*/) {
   // Metadata Frame is not supported in QUIC.
   // TODO(danzh): add stats for metadata not supported error.
+  ENVOY_STREAM_LOG(error, "METADATA is not supported in Http3.", *this);
```

Comment:
ENVOY_BUG would be appropriate in this case.

Reply:
I'd prefer we stick with stat and log, as we do for H1, so I think this is fine as-is